tdrs-backend/tdpservice/parsers/models.py

```diff
@@ -67,3 +67,71 @@ def __str__(self):
     def _get_error_message(self):
         """Return the error message."""
         return self.error_message
+
+class DataFileSummary(models.Model):
+    """Aggregates information about a parsed file."""
+
+    class Status(models.TextChoices):
+        """Enum for status of parsed file."""
+
+        PENDING = "Pending"  # file has been uploaded, but not validated
+        ACCEPTED = "Accepted"
+        ACCEPTED_WITH_ERRORS = "Accepted with Errors"
+        REJECTED = "Rejected"
+
+    status = models.CharField(
+        max_length=50,
+        choices=Status.choices,
+        default=Status.PENDING,
+    )
+
+    datafile = models.ForeignKey(DataFile, on_delete=models.CASCADE)
+
+    #  eventually needs a breakdown of cases (accepted, rejected, total) per month
+    #  elif qtr2 jan-mar
+    #  elif qtr3 apr-jun
+    #  elif qtr4 jul-sept
+
+    case_aggregates = models.JSONField(null=True, blank=False)
+    """
+    # Do these queries only once, save result during creation of this model
+    # or do we grab this data during parsing and bubble it up during create call?
+    {
+        "Jan": {
+            "accepted": 100,
+            "rejected": 10,
+            "total": 110
+        },
+        "Feb": {
+            "accepted": 100,
+            "rejected": 10,
+            "total": 110
+        },
+        "Mar": {
+            "accepted": 100,
+            "rejected": 10,
+            "total": 110
+        }
+    """
+
+    def set_status(self, errors):
+        """Set and return the status field based on errors and models associated with datafile."""
+        # to set rejected, we would need to have raised an exception during (pre)parsing
+        # else if there are errors, we can set to accepted with errors
+        # else we can set to accepted (default)
+        if errors == {}:  # This feels better than running len() on `errors`...but is it a dict vs list?
+            self.status = self.Status.ACCEPTED
+        elif check_for_preparsing(errors):
+            self.status = self.Status.REJECTED
+        elif errors:
+            self.status = self.Status.ACCEPTED_WITH_ERRORS
+
+        return self.status
+
+def check_for_preparsing(errors):
+    """Check for pre-parsing errors."""
+    for key in errors.keys():  # keys are 'header', 'trailer', 'document'
+        for error in errors[key]:
+            if error.category == "1":
```

Comment:
TODO: utilize Jan's new `choices` both here and in the parserError factory.

Reply:
TODO: comments for expected structure for errors since keys might also include line_number